Source/Kesmai.Server/Game/Items/Equipment/Shields/Shield.cs

```diff
@@ -54,18 +55,74 @@ protected Shield(int shieldID) : base(shieldID)
 		{
 		}
 
+		public int GetMeleeBonus(ItemEntity item, WeaponFlags flags)
+		{
+			var protectionBonus = 0;
+			var itemProtections = new List<int>();
+
+			if ((flags & WeaponFlags.Piercing) != 0)
+				itemProtections.Add(PiercingProtection);
+
+			if ((flags & WeaponFlags.Slashing) != 0)
+				itemProtections.Add(SlashingProtection);
+
+			if ((flags & WeaponFlags.Bashing) != 0)
+				itemProtections.Add(BashingProtection);
+
+			if (itemProtections.Any())
+				protectionBonus += itemProtections.Min();
+
+			return protectionBonus;
+		}
+		
 		/// <inheritdoc/>
 		/// <remarks>
 		/// Shields only provide a blocking bonus against weapons when equipped in the left-hand.
+		/// Used for DR calculations.
 		/// </remarks>
 		public override int CalculateBlockingBonus(ItemEntity item)
 		{
-			if (item is IWeapon weapon && weapon.Flags.HasFlag(WeaponFlags.Projectile))
-				return ProjectileProtection;
+			var flags = WeaponFlags.Bashing;
+
+			if (item is IWeapon weapon)
+				flags = weapon.Flags;
 
-			return BaseArmorBonus;
+			var bonusBlock = 0;
+			/// Let shields use BaseArmorBonus to avoid projectiles	
+			if ((flags & WeaponFlags.Projectile) != 0)
+				bonusBlock += ProjectileProtection;
+			else
+				bonusBlock += GetMeleeBonus(item, flags);
+
+			return BaseArmorBonus + bonusBlock;
 		}
-		
+
+		/// <inheritdoc/>
+		/// <remarks>
+		/// Shields only provide a blocking bonus against weapons when equipped in the left-hand.
+		/// Used for mitigation calculations.
+		/// </remarks>
+		public int GetShieldBonus(ItemEntity item)
+		{
+			var flags = WeaponFlags.Bashing;
+
+			if (item is IWeapon weapon)
+				flags = weapon.Flags;
+
+			var shieldBonus = 0;
+
+			if ((flags & WeaponFlags.Projectile) != 0)
+			{
+				shieldBonus += ProjectileProtection;
+			}
+			else
+			{
+				shieldBonus += GetMeleeBonus(item, flags);
+			}
+
+			return shieldBonus + BaseArmorBonus;
```

Comment:
Name refactoring still seems a bit off but I believe this should not add baseArmorBonus